File: pulsar/java/src/main/java/com/examples/hello/pulsar/producers/ProducerController.java
package com.examples.hello.pulsar.producers;

import com.examples.hello.pulsar.PulsarExecutors;
import com.examples.hello.pulsar.ServiceController;
import io.micronaut.http.annotation.Controller;
import jakarta.inject.Inject;
import jakarta.inject.Named;
import lombok.Getter;

import java.util.concurrent.ExecutorService;

@Controller("/producer")
public class ProducerController extends ServiceController<ProducerRunnable> {
  @Named(PulsarExecutors.PRODUCER)
  @Getter @Inject
  ExecutorService executor;
  @Getter @Inject
  ProducerRunnable service;
}

File: pulsar/kotlin/src/main/kotlin/com/example/hello/pulsar/producers/ProducerController.kt
package com.example.hello.pulsar.producers

import com.example.hello.pulsar.PulsarExecutors
import com.example.hello.pulsar.ServiceController
import io.micronaut.http.annotation.Controller
import jakarta.inject.Inject
import jakarta.inject.Named
import java.util.concurrent.ExecutorService

@Controller("/producer")
class ProducerController(
  @Inject @Named(PulsarExecutors.PRODUCER) val executor: ExecutorService,
  @Inject val service: BatchProducerRunnable
) : ServiceController<ProducerRunnable>(executor, service)
